drivers/syslog/syslog_device.c

```diff
@@ -435,22 +435,44 @@ void syslog_dev_uninitialize(FAR struct syslog_channel_s *channel)
 {
   FAR struct syslog_dev_s *syslog_dev = (FAR struct syslog_dev_s *)channel;
 
-  /* Check if the system is ready */
+  /* Uninitializing a SYSLOG device should not take place within
+   * interrupt context.
+   */
 
-  if (syslog_dev_outputready(syslog_dev) < 0)
+  if (up_interrupt_context() || getpid() == 0)
     {
+      DEBUGASSERT(!up_interrupt_context() && getpid() != 0);
       return;
     }
 
-  /* Attempt to flush any buffered data */
+  /* The device cannot be unititialized while it is being
```

Comment:
```suggestion
  /* The device cannot be uninitialized while it is being
```

Reply:
Good catch, I was staring at this a couple of minutes, trying to understand the difference...